presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -2052,7 +2072,9 @@ else if (fileStatus.isDirectory()) {
                 notDeletedEligibleItems.add(fileStatus.getPath().toString());
             }
         }
-        if (allDescendentsDeleted && deleteEmptyDirectories) {
+        // Unconditionally delete empty delta_ and delete_delta_ directories, because that's
+        // what Hive does, and leaving them in place confuses delta file readers.
```

Comment:
The ORC reader see the empty delta directory, and then throws an exception because it can't find the _orc_acid_version file in that directory.  And Hive never leaves empty delta directories around.

Reply:
Thank for explaining.

I think it's worth replacing "confuses" with something indicating a failure.
(Confuse may imply "failure", "incorrect results", "performance penalty", ...)

Also, worth mentioning which Hive version this applies to. (the world is in flux)

Maybe

```
// Unconditionally delete empty delta_ and delete_delta_ directories.
// Hive 3.1 fails when it finds a delta_ OR delete_delta_ directory without _orc_acid_version file.
```